atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/paxos/PaxosServerImplementation.java

```diff
@@ -0,0 +1,233 @@
+        }
+
+        registerLeaderElectionService(configuration);
+        environment.jersey().register(new NotCurrentLeaderExceptionMapper());
+    }
+
+    private void registerPaxosResource() {
+        paxosResource = PaxosResource.create(paxosConfiguration.paxosDataDir().toString());
+        paxosResource.addClient(LEADER_NAMESPACE);
+        environment.jersey().register(paxosResource);
+    }
+
+    private void registerLeaderElectionService(TimeLockServerConfiguration configuration) {
+        remoteServers = getRemotePaths(configuration);
+        List<PaxosLearner> learners = getNamespacedProxies(
+                remoteServers,
+                LEADER_NAMESPACE,
+                optionalSecurity,
+                PaxosLearner.class);
+        learners.add(paxosResource.getPaxosLearner(LEADER_NAMESPACE));
+
+        List<PaxosAcceptor> acceptors = getNamespacedProxies(
+                remoteServers,
+                LEADER_NAMESPACE,
+                optionalSecurity,
+                PaxosAcceptor.class);
+        acceptors.add(paxosResource.getPaxosAcceptor(LEADER_NAMESPACE));
+
+        Map<PingableLeader, HostAndPort> otherLeaders = Leaders.generatePingables(
+                remoteServers,
+                optionalSecurity);
+
+        ExecutorService executor = Executors.newCachedThreadPool(new ThreadFactoryBuilder()
+                .setNameFormat("atlas-leaders-%d")
+                .setDaemon(true)
+                .build());
+
+        PaxosProposer proposer = PaxosProposerImpl.newProposer(
+                paxosResource.getPaxosLearner(LEADER_NAMESPACE),
+                ImmutableList.copyOf(acceptors),
+                ImmutableList.copyOf(learners),
+                getQuorumSize(configuration),
+                executor);
+
+        // Build and store a gatekeeping leader election service
+        leaderElectionService = new PaxosLeaderElectionService(
+                proposer,
+                paxosResource.getPaxosLearner(LEADER_NAMESPACE),
+                otherLeaders,
+                ImmutableList.copyOf(acceptors),
+                ImmutableList.copyOf(learners),
+                executor,
+                paxosConfiguration.pingRateMs(),
+                paxosConfiguration.randomWaitBeforeProposalMs(),
+                paxosConfiguration.leaderPingResponseWaitMs());
+        environment.jersey().register(leaderElectionService);
+    }
+
+    private static int getQuorumSize(TimeLockServerConfiguration configuration) {
+        return configuration.cluster().servers().size() / 2 + 1;
+    }
+
+    private static Optional<SSLSocketFactory> constructOptionalSslSocketFactory(
+            PaxosConfiguration configuration) {
+        return Optional.of(SslSocketFactories.createSslSocketFactory(configuration.sslConfiguration().get()));
+    }
+
+    @Override
+    public void onStop() {
+        // Nothing to do
+    }
+
+    @Override
+    public void onStartupFailure() {
+        // Nothing
```

Comment:
This is called in `TimeLockServer`, when startup throws an exception. This would be important if we create additional external services. (This was put in to address #1282; that issue would also apply to Timelock failing whilst starting up).

For Paxos we don't have any "other" services that need to be shut down, in that sense, so there is no work here. I've given the `onStop()` and `onStartupFailure()` no-op defaults in `TimeLockServerImplementation`, and removed these empty impls.